packages/directlinespeech/src/DirectLineSpeech.js

```diff
@@ -0,0 +1,108 @@
+/* eslint no-magic-numbers: ["error", { "ignore": [0, 1, 2, 36] }] */
+
+import Observable from 'core-js/features/observable';
+
+import shareObservable from './shareObservable';
+import { fromAudioStream as createUtteranceFromAudioStream } from './SpeechSynthesisAudioStreamUtterance';
+
+function randomActivityId() {
```

Comment:
Can you explain this function?

Reply:
RxJS 5 behaves differently than ES Observable.

In ES Observable, when someone subscribe to an Observable, it creates a new subscription. For example,

```js
const observable = new Observable(observer => {
  console.log('subscribed');
});

observable.subscribe();
observable.subscribe();
```

You will see 2 x `subscribed` message.

But in RxJS 5, it's "shared". That means, you will only see 1 x `subscribed` message.

In DLJS and DLSpeech, it will establish the connection only when the first subscriber has subscribed. When multiple subscribers subscribe, all of them will share the same subscription, and the first subscriber will establish the connection.

As we are using ES Observable in our DLSpeech code, we need to mimic this behavior for Web Chat. Thus, `shareObservable` is to create a new ES Observable that multiplex multiple subscriptions into a single subscription.

Reply:
(Please resolve this one after you read the explanation.)

Reply:
I'm confused, I was asking about `randomActrivityId` xD